sklearn/tree/_criterion.pxd

```diff
@@ -49,10 +49,20 @@ cdef class Criterion:
     cdef double* sum_left           # Same as above, but for the left side of the split
     cdef double* sum_right          # same as above, but for the right side of the split
 
+    cdef double* coefficients
+
+    cdef DOUBLE_t* lift_se
```

Comment:
Could we also make it less specific to our use-case? Can we reuse the weights variable?